core/sr-primitives/src/lib.rs

```diff
@@ -237,6 +237,88 @@ where
 	}
 }
 
+/// These implementation will never panic and will always return a _per-thing_ which is in the range
+/// `[0, 1] (inclusive)`.
+impl Saturating for Permill {
+	fn saturating_add(self, rhs: Self) -> Self {
+		// defensive inner saturation:
+		// both u32 values are below a million. sum won't overflow in u32.
+		Self::from_parts((self.0 + rhs.0).min(1_000_000))
```

Comment:
the min is redundant with inner min in from_parts but I'm fine to keep it as well

Reply:
Yeah remove it.